crates/chain/src/tx_data_traits.rs

```diff
@@ -44,6 +44,55 @@ impl ForEachTxOut for Transaction {
 /// assume that transaction A is also confirmed in the best chain. This does not necessarily mean
 /// that transaction A is confirmed in block B. It could also mean transaction A is confirmed in a
 /// parent block of B.
+///
+/// ```
+/// # use bdk_chain::local_chain::LocalChain;
+/// # use bdk_chain::tx_graph::TxGraph;
+/// # use bdk_chain::BlockId;
+/// # use bdk_chain::ConfirmationHeightAnchor;
+/// # use bdk_chain::example_utils::*;
+/// # use bitcoin::hashes::Hash;
+///
+/// // Initialize the local chain with two blocks.
+/// let chain = LocalChain::from_blocks(
+///     [
+///         (1, Hash::hash("first".as_bytes())),
+///         (2, Hash::hash("second".as_bytes())),
+///     ]
+///     .into_iter()
+///     .collect(),
+/// );
+///
+/// // Insert transaction into TxGraphs with different anchor types.
+/// // Insert different anchor types for the same transaction.
+/// // When using the BlockId as anchor, the anchor block is the
+/// // confirmation block of the transaction. Where as when using
+/// // the ConfirmationHeightAnchor, the anchor block is a bound on
+/// // the confirmation block of the transaction.
```

Comment:
nit: maybe we should explicitly say in the ConfirmationHeightAnchor (and ConfirmationTimeAnchor) docs that being anchored to a block doesn't mean that the transaction is confirmed in that block?

Reply:
Also, what do you mean by "the anchor block is *a bound on* the confirmation block"? I think it could be rephrased in a clearer way.

Reply:
I have updated the docs for ConfirmationHeightAnchor and ConfirmationTimeAnchor. I have also removed the confusing clause.